src/nvc.rs

```diff
@@ -155,10 +257,19 @@ impl Nvc {
         self.psw_interrupt_mask_level = ((value as usize) >> 16) & 0x0f;
     }
 
+    pub fn fetch_instruction(&mut self, interconnect: &mut Interconnect, addr: u32) -> (u16, bool) {
+        let halfword = interconnect.read_halfword(addr);
```

Comment:
So i'm clear, you are suggesting this instead of the fetch_instruction() stuff?

let (first_halfword, _) = self.cache.read_halfword(interconnect, pc_original);